api/services/MachineService.js

```diff
@@ -508,20 +508,21 @@ function updateMachinesPool() {
 function _shouldTerminateMachine(machine) {
   Promise.props({
     isActive: machine.isSessionActive(),
-    config: ConfigService.get('neverTerminateMachine')
+    config: ConfigService.get('neverTerminateMachine'),
+    machineToTerminate: Machine.findOne({id: machine.id})
```

Comment:
No, machine is the not updated machine, if you reopen/reclose the vdi, the machine you received is not updated with the right endDate. Search the machine with findOne, permit to have the right endDate.